src/Drupal/DrupalExtension/Context/RawDrupalContext.php

```diff
@@ -451,22 +503,23 @@ public function languageCreate(\stdClass $language) {
   }
 
   /**
-   * Log-in the current user.
+   * Log-in the given user.
+   *
+   * @param \stdClass $user
+   *   The user to log in.
    */
-  public function login() {
+  public function login(\stdClass $user) {
```

Comment:
I changed the way this works because the way this used to work briefly left the user in an inconsistent state. We used to first populate the `$this->user` property, and then call `$this->login()`. This meant the "current user" was already set before the user was even logged in.

It seems more logical and consistent to me to actually set the current user after the user has been successfully logged in.